pandas/core/strings.py

```diff
@@ -1095,24 +1095,83 @@ def str_pad(arr, width, side='left', fillchar=' '):
 
 def str_split(arr, pat=None, n=None):
     """
-    Split each string (a la re.split) in the Series/Index by given
-    pattern, propagating NA values. Equivalent to :meth:`str.split`.
+    Split strings around given separator/delimiter.
+
+    Split each str in the caller's values by given
+    pattern, propagating NaN values. Equivalent to :meth:`str.split`.
 
     Parameters
     ----------
     pat : string, default None
-        String or regular expression to split on. If None, splits on whitespace
+        String or regular expression to split on.
+        If `None`, split on whitespace.
     n : int, default -1 (all)
-        None, 0 and -1 will be interpreted as return all splits
+        Vary dimensionality of output.
+
+        * `None`, 0 and -1 will be interpreted as return all splits
     expand : bool, default False
-        * If True, return DataFrame/MultiIndex expanding dimensionality.
-        * If False, return Series/Index.
+        Expand the split strings into separate columns.
 
-    return_type : deprecated, use `expand`
+        * If `True`, return DataFrame/MultiIndex expanding dimensionality.
+        * If `False`, return Series/Index.
 
     Returns
     -------
+    Type matches caller unless `expand=True` (return type is `DataFrame`)
```

Comment:
return type for expand=True can also be MultiIndex I think

Reply:
I think the wording around `Index` and `MultiIndex` is confusing here - how does this return a `MultiIndex`?

Reply:
Ah, sorry @WillAyd, merged maybe a bit too quick

When you use this on an index, and do `expand=True`, you get a MultiIndex (similar as if Series -> DataFrame). 

Maybe it would also be good to add an example of this. 
@mananpal1997 Feel free to open a new PR to do this small follow-up change.

Reply:
Ah OK makes sense. Yes I just find the text like "Series/Index" and "DataFrame/MultiIndex" to be rather confusing - part of me wants to interpret those as being bound together, so that a Series is always indexed normally and a DataFrame is indexed with a MultiIndex. Obviously that's not the case, but I think that we could more clearly delineate.

@mananpal1997 an example would certainly help for your next contribution!

Reply:
I'll add it :+1:

Reply:
@WillAyd @jorisvandenbossche 
Any edit suggestions to this change before I push it?
```
Parameter ``expand=True`` returns DataFrame and MultiIndex objects for
    Series and Index objects respectively.

    >>> i = pd.Index(["ba 100 001", "ba 101 002", "ba 102 003"])
    >>> type(s.str.split(expand=True))
    <class 'pandas.core.frame.DataFrame'>
    >>> type(i.str.split(expand=True))
    <class 'pandas.core.indexes.multi.MultiIndex'>
```

Reply:
You can ping us both in the new PR, so let's further discss there. 
But I think I would just show the actual result, instead of the types, it will be clear from the result what the different types are